ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/peers/EnodeURL.java

```diff
@@ -181,18 +195,33 @@ private OptionalInt getDiscPortQueryParam() {
   }
 
   public static URI asURI(final String url) {
-    return fromString(url).toURI();
+    return asURI(url, EnodeDnsConfiguration.dnsDisabled());
+  }
+
+  public static URI asURI(final String url, final EnodeDnsConfiguration enodeDnsConfiguration) {
+    return fromString(url, enodeDnsConfiguration).toURI();
   }
 
   public Bytes getNodeId() {
     return nodeId;
   }
 
   public String getIpAsString() {
-    return ip.getHostAddress();
+    return getIp().getHostAddress();
   }
 
   public InetAddress getIp() {
+    this.ip =
```

Comment:
Is doing a DNS lookup every time the design intent?  If not perhaps `ip` should be a supplier and this logic gets put in a `Suppliers.memoize` call?  If it is the design intent it's work a comment so a later refactoring doesn't undo it.

Reply:
No it's a voluntary choice to DNS lookup every time. Because a cache could give a bad IP before it is invalidated again. If the "dns-update-enabled" flag is false it will do it only once. 
we could add a cache but I'm afraid that this will prevent nodes from communicating each other while waiting for the next cache update

Reply:
I just added a comment 👍